library/src/main/java/com/readystatesoftware/chuck/internal/data/entity/HttpTransaction.kt

```diff
@@ -0,0 +1,205 @@
+
+/**
+ * Represent a full HTTP transaction (with Request and Response). Instances of this classes
+ * should be populated as soon as the library receives data from OkHttp.
+ */
+@Entity(tableName = "transactions")
+internal class HttpTransaction(
+        @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "id") var id: Long = 0,
+        @ColumnInfo(name = "requestDate") var requestDate: Long?,
+        @ColumnInfo(name = "responseDate") var responseDate: Long?,
+        @ColumnInfo(name = "tookMs") var tookMs: Long?,
+        @ColumnInfo(name = "protocol") var protocol: String?,
+        @ColumnInfo(name = "method") var method: String?,
+        @ColumnInfo(name = "url") var url: String?,
+        @ColumnInfo(name = "host") var host: String?,
+        @ColumnInfo(name = "path") var path: String?,
+        @ColumnInfo(name = "scheme") var scheme: String?,
+        @ColumnInfo(name = "requestContentLength") var requestContentLength: Long?,
+        @ColumnInfo(name = "requestContentType") var requestContentType: String?,
+        @ColumnInfo(name = "requestHeaders") var requestHeaders: String?,
+        @ColumnInfo(name = "requestBody") var requestBody: String?,
+        @ColumnInfo(name = "isRequestBodyPlainText") var isRequestBodyPlainText: Boolean = true,
+        @ColumnInfo(name = "responseCode") var responseCode: Int?,
+        @ColumnInfo(name = "responseMessage") var responseMessage: String?,
+        @ColumnInfo(name = "error") var error: String?,
+        @ColumnInfo(name = "responseContentLength") var responseContentLength: Long?,
+        @ColumnInfo(name = "responseContentType") var responseContentType: String?,
+        @ColumnInfo(name = "responseHeaders") var responseHeaders: String?,
+        @ColumnInfo(name = "responseBody") var responseBody: String?,
+        @ColumnInfo(name = "isResponseBodyPlainText") var isResponseBodyPlainText: Boolean = true
+
+) {
+
+    @Ignore
+    constructor() : this(
+            requestDate = null,
+            responseDate = null,
+            tookMs = null,
+            protocol = null,
+            method = null,
+            url = null,
+            host = null,
+            path = null,
+            scheme = null,
+            requestContentLength = null,
+            requestContentType = null,
+            requestHeaders = null,
+            requestBody = null,
+            responseCode = null,
+            responseMessage = null,
+            error = null,
+            responseContentLength = null,
+            responseContentType = null,
+            responseHeaders = null,
+            responseBody = null)
+
+    enum class Status {
+        Requested,
+        Complete,
+        Failed
+    }
+
+    val status: Status
+        get() = when {
+            error != null -> Status.Failed
+            responseCode == null -> Status.Requested
+            else -> Status.Complete
+        }
+
+    val requestDateString: String?
+        get() = requestDate?.toString()
```

Comment:
So this can easily be removed. The problem is unfortunately wider.
So far I tried to make sure `HttpTransaction.kt` had a similar api with `HttpTransaction.java`.

I'll say we can probably refactor all this stuff outside of the data class and have it inside say FormatUtils?
I'll anyway keep those refactorings outside of this PR.